### PR TITLE
6939 add sysevents to zfs core for commands

### PR DESCRIPTION
**Pull request by bahamas10**
once this commit looks good it will be force-pushed to the branch that is currently in PR with the OpenZFS project (https://github.com/openzfs/openzfs/pull/101)

### CODE REVIEW

File: usr/src/uts/common/fs/zfs/spa_history.c [hunk 1]
```diff
@@ -255,6 +266,22 @@ spa_history_log_sync(void *arg, dmu_tx_t *tx)
...
+		 * The history sysevent is posted only for internal history
+		 * messages to show what has happened, not how it happened. For
+		 * example, the following command:
+		 *
+		 * # zfs destroy -r tank/foo
+		 *
+		 * will result in one sysevent posted per dataset that is
+		 * destroyed as a result of the command - which could be more
+		 * than one event in total.  By contrast, if the sysevent was
+		 * posted as a result of the ZPOOL_HIST_CMD key being present
```
Comment (pfmooney) [inline usr/src/uts/common/fs/zfs/spa_history.c]: append comma to 'present'

File: usr/src/uts/common/fs/zfs/spa.c [hunk 2]
```diff
@@ -6985,6 +6987,14 @@ spa_event_create(spa_t *spa, vdev_t *vd, const char *name)
 		}
 	}
 
+	if (hist_nvl) {
```
Comment (pfmooney) [inline usr/src/uts/common/fs/zfs/spa.c]: I would prefer a more explicit '!= NULL' comparison, but perhaps this matches the local style expectations.

**Review by pfmooney — COMMENTED** [review]
Here are a couple nits.  Otherwise it looks OK to me.

File: usr/src/uts/common/fs/zfs/spa.c (hunk 2 above)
Comment (bahamas10) [reply usr/src/uts/common/fs/zfs/spa.c]: I would actually prefer that as well, and only did it like this to match the style already in place.  If you expand the lines above you can see `if (vd) { ... }` already in place with almost the exact same logic as this block.

File: usr/src/uts/common/fs/zfs/spa_history.c (hunk 1 above)
Comment (bahamas10) [reply usr/src/uts/common/fs/zfs/spa_history.c]: agreed, will update.

File: usr/src/uts/common/fs/zfs/spa.c (hunk 2 above)
Comment (pfmooney) [reply usr/src/uts/common/fs/zfs/spa.c]: I think you could make the call either way.  I'm satisfied with whatever you decide.

Comment (bahamas10) [reply usr/src/uts/common/fs/zfs/spa.c]: after review from joshc bringing up the same issue, I agree that it should be `if (hist_nvl !=null) { ... }` as it explicitly conveys our intent - will change.

**Pull request closed without merging**